### PR TITLE
Add main resources

### PR DESCRIPTION
**Pull request by alexandra-buica**
Description

Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring (no functional changes, formatting, renaming)
- [X] Other (please describe): add main resources to the module

TODOs

- [X] Ensure the branch is named correctly with the ticket number. (eg: DOP-2222)
- [X] Update the docs.
- [X] Run the pre-commit checks successfully.

Other information

Adding reviewers:
@coremaker/devops

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  type = map(object({
+    partitions_count = number
+    config           = map(string)
+  }))
+  default     = {}
+  description = "This helps to create a Topic resource along with the specific settings."
+}
+
+# API Keys
+variable "services" {
```
Comment (CosminBriscaru) [inline variables.tf]: can we rename this var to something more meaningful?
ex: services_acls, iam_services

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# ENVIRONMENT
+variable "confluent_environment_name" {
```
Comment (CosminBriscaru) [inline variables.tf]: Can we name this one just environment_name?
it is a module specially designed for confluent anyway.

File: topics-iam.tf [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  ])
+  write_topic_pairs = flatten([
+    for service in var.services : [
+      for topic in service.writeTopics : {
+        topic = topic
+        name  = service.name
+      }
+    ]
+  ])
+  services = {
```
Comment (CosminBriscaru) [inline topics-iam.tf]: can we rename this local var to something more meaningful?
ex: services_acls, iam_services

**Review by CosminBriscaru — CHANGES_REQUESTED** [review]

**Comment by mihneastaub** [comment]
Is looking great, but I think is better to have also a directory with examples. Something like this: https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/examples/complete.

You can create a directory in the root repository with the name `examples` and there you can add the configuration files.

File: README.md [hunk 4]
```diff
@@ -1 +1,114 @@
...
+[![Maintained by coremaker.io](https://img.shields.io/badge/maintained%20by-coremaker.io-green)](https://coremaker.io/)
+
+# Terraform-confluent-cloud
+
+This is a terraform module for Confluent Cloud provider which helps you build environments, clusters, topics and API Keys with restricted access on ACLs.  
+
+# Examples
+
+```terraform
+module "kubernetes" {
```
Comment (mihneastaub) [inline README.md]: Please change the name of this module.

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  default     = {}
+  description = "This helps to create a Topic resource along with the specific settings."
+}
+
+# API Keys
+variable "services_acls" {
+  type = list(object({
+    name        = string
+    readTopics  = list(string)
+    writeTopics = list(string)
```
Comment (CosminBriscaru) [inline variables.tf]: let's mark readTopics and writeTopics  as optional, terraform 1.3.0 version.

File: variables.tf (hunk 1 above)
Comment (alexandra-buica) [reply variables.tf]: Variable renamed.

File: variables.tf (hunk 2 above)
Comment (alexandra-buica) [reply variables.tf]: Done.

**Review by CosminBriscaru — APPROVED** [review]
LGTM

**Pull request merged**